src/autowiring/test/AutoConfigTest.cpp

```diff
@@ -5,11 +5,37 @@
 #include <autowiring/ConfigRegistry.h>
 #include <autowiring/observable.h>
 
+namespace aw = autowiring;
+
 class AutoConfigTest:
   public testing::Test
 {};
 
 namespace {
+  class multi_meta {
+  public:
+    static const bool multi = true;
+
+    std::string name;
+
+    template<typename T>
+    void bind(const aw::config_field& field, T&) {
+      name = field.name;
+    }
+  };
+
+  static_assert(aw::valid<int, multi_meta>::value, "Float metadata field not detected as being valid");
+
+  class single_meta {
+  public:
+    single_meta(const char* name) :
+      name(name)
+    {}
+
+    const char* name;
+    static const bool multi = false;
```

Comment:
Since the absence of a multi definition in a type appears to yield "is_multi<type>::value == false" why not demonstrate this by removing the definition here and comment accordingly what the expectations are?